packages/taiko-client/proposer/proposer.go

```diff
@@ -235,21 +235,40 @@ func (p *Proposer) fetchPoolContent(allowEmptyPoolContent bool) ([]types.Transac
 // from L2 execution engine's tx pool, splitting them by proposing constraints,
 // and then proposing them to TaikoInbox contract.
 func (p *Proposer) ProposeOp(ctx context.Context) error {
-	// Check if the preconfirmation router is set, if so, skip proposing.
-	preconfRouter, err := p.rpc.GetPreconfRouterPacaya(&bind.CallOpts{Context: ctx})
-	if err != nil {
-		return fmt.Errorf("failed to fetch preconfirmation router: %w", err)
-	}
-	if preconfRouter != rpc.ZeroAddress {
-		log.Info("Preconfirmation router is set, skip proposing", "address", preconfRouter, "time", time.Now())
-		return nil
-	}
-
 	// Wait until L2 execution engine is synced at first.
 	if err := p.rpc.WaitTillL2ExecutionEngineSynced(ctx); err != nil {
 		return fmt.Errorf("failed to wait until L2 execution engine synced: %w", err)
 	}
 
+	// Check if the preconfirmation router is set, if so, skip proposing.
+	if p.rpc.PacayaClients.PreconfRouter != nil {
+		fallbackPreconferAddress, err := p.rpc.PacayaClients.PreconfRouter.FallbackPreconfer(&bind.CallOpts{Context: ctx})
+		if err != nil {
+			return fmt.Errorf("failed to get fallback preconfer address: %w", err)
+		}
+		if fallbackPreconferAddress != p.proposerAddress || fallbackPreconferAddress != p.ProverSetAddress {
+			log.Info("Preconfirmation is activated and proposer isn't the fallback preconfer, skip proposing",
+				"time",
+				time.Now(),
+			)
+			return nil
+		}
+		// As a fallback preconfer, need to enable preconfirmation server
```

Comment:
we are not a fallback preconfer, we are basically going to just be a fallback proposer.